### PR TITLE
Sap collector pr

### PR DESCRIPTION
**Pull request by snorkyware**

### CODE REVIEW

File: ValheimPlus/GameClasses/SapCollector.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            }
+        }
+    
+        /// <summary>
+        /// Auto Deposit for SapCollectors
+        /// </summary>
+        [HarmonyPatch(typeof(SapCollector), nameof(SapCollector.RPC_Extract))]
+        public static class SapCollector_RPC_Extract_Patch
+        {
+            private static bool Prefix(long caller, SapCollector __instance)
```
Comment (Grantapher) [inline ValheimPlus/GameClasses/SapCollector.cs:49]: I don't believe the first parameter is necessary.

File: ValheimPlus/GameClasses/SapCollector.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        public static class SapCollector_UpdateTick_Patch
+        {
+            private static void Postfix(SapCollector __instance)
+            {
+                if (__instance.GetLevel() != __instance.m_maxLevel) return;
+                Deposit(__instance);
+            }
+        }
+
+        private static bool Deposit(SapCollector __instance)
```
Comment (Grantapher) [inline ValheimPlus/GameClasses/SapCollector.cs:66]: Does this always return true? If so, then you can just have it return void.

File: ValheimPlus/GameClasses/SapCollector.cs (hunk 1 above)
Comment (snorkyware) [reply ValheimPlus/GameClasses/SapCollector.cs:49]: It's not. I missed it when copying from the Beehive game class, where it is also unnecessary.

File: ValheimPlus/GameClasses/SapCollector.cs (hunk 2 above)
Comment (snorkyware) [reply ValheimPlus/GameClasses/SapCollector.cs:66]: It does not.

**Comment by Grantapher** [comment]
Committed in 5864508.

**Pull request closed without merging**